### PR TITLE
Set AutoPanMode

### PR DESCRIPTION
**Pull request by sbergmair**
This PR adds following functionality:
- setAutoPanMode: Enables setting the AutoPanMode like `off`, `recenter`, `navigation`, `compassNavigation` (https://developers.arcgis.com/net/api-reference/api/netwin/Esri.ArcGISRuntime/Esri.ArcGISRuntime.UI.LocationDisplay.AutoPanMode.html)
- getAutoPanMode: expose the current set value
- setWanderExtentFactor: This is to configure how fast the map gets centered in `AutoPanMode.recenter` (https://developers.arcgis.com/net/api-reference/api/netwin/Esri.ArcGISRuntime/Esri.ArcGISRuntime.UI.LocationDisplay.WanderExtentFactor.html#Esri_ArcGISRuntime_UI_LocationDisplay_WanderExtentFactor)
- getWanderExtentFactor: expose the current set value

information for setWanderExtentFactor: If autoPanMode is not `recenter`, there is no effect, which is intended.

### CODE REVIEW

File: arcgis_map_sdk_android/android/src/main/kotlin/dev/fluttercommunity/arcgis_map_sdk_android/ArcgisMapView.kt [hunk 1]
```diff
@@ -263,6 +268,71 @@ internal class ArcgisMapView(
...
+        } catch (e: Throwable) {
+            result.finishWithError(e, "Setting AutoPanMode failed.")
+        }
+    }
+
+    private fun onGetAutoPanMode(
+        call: MethodCall,
+        result: MethodChannel.Result
+    ) {
+        return result.success(mapView.locationDisplay.autoPanMode.name)
```
Comment (JulianBissekkou) [inline arcgis_map_sdk_android/android/src/main/kotlin/dev/fluttercommunity/arcgis_map_sdk_android/ArcgisMapView.kt:305]: I think arcgis is a java sdk and we might have a property that is not correctly annotated with `@Nullable`. In that case we would even get a null pointer.

I am almost sure that this is not the case, but lets double check 👍🏽 
You can also wrap this in a try catch just to be save.

File: arcgis_map_sdk_ios/ios/Classes/ArcgisMapView.swift [hunk 2]
```diff
@@ -710,3 +748,37 @@ extension AGSLoadStatus {
...
+        case .off:
+            return "off"
+        case .recenter:
+            return "recenter"
+        case .navigation:
+            return "navigation"
+        case .compassNavigation:
+            return "compassNavigation"
+        @unknown default:
+            return "unknown"
```
Comment (JulianBissekkou) [inline arcgis_map_sdk_ios/ios/Classes/ArcgisMapView.swift:781]: lets default to `nil` and fallback like you did in kotlin. I really liked that :D

File: arcgis_map_sdk_method_channel/lib/src/method_channel_arcgis_map_plugin.dart [hunk 3]
```diff
@@ -34,6 +34,36 @@ class MethodChannelArcgisMapPlugin extends ArcgisMapPlatform {
...
+  }
+
+  @override
+  Future<AutoPanMode> getAutoPanMode(int mapId) {
+    return _methodChannelBuilder(mapId)
+        .invokeMethod<String>("get_auto_pan_mode")
+        .then(
+          (value) =>
+              AutoPanMode.values.firstWhere((element) => element.name == value),
+        );
```
Comment (JulianBissekkou) [inline arcgis_map_sdk_method_channel/lib/src/method_channel_arcgis_map_plugin.dart:49]: Does this work?
```suggestion
        .then(AutoPanMode.values.byName);
```

**Review by JulianBissekkou — CHANGES_REQUESTED** [review]
Almost perfect! 💪🏽 
Can you also update the readme with the supported features?

File: arcgis_map_sdk_method_channel/lib/src/method_channel_arcgis_map_plugin.dart (hunk 3 above)
Comment (sbergmair) [reply arcgis_map_sdk_method_channel/lib/src/method_channel_arcgis_map_plugin.dart:49]: `(value)` is nullable. I need to fallback in this case 🤔

**Review by JulianBissekkou — APPROVED** [review]

**Pull request merged**